packages/cli/generation/ir-generator/src/generateIntermediateRepresentation.ts

```diff
@@ -263,7 +268,9 @@ export async function generateIntermediateRepresentation({
         }
     });
 
-    const filteredIr = audienceIrGraph != null ? audienceIrGraph.build() : undefined;
+    intermediateRepresentation.typeGraph = extractTypeGraph(irGraph);
```

Comment:
`extractTypeGraph` is not the greatest name for the method, i find myself asking wtf is this doing. ideally we should get a high level overview of whats happening just by reading the method name

Reply:
https://github.com/fern-api/fern/pull/1993/files#r1281193699 if we do this rename then i could se this being `computeServiceTypeReferenceInfo` and that gives me a little more info

Reply:
Two options:
1. `newTypeGraphFromIrGraph`
2. Add an `IrGraph.toTypeGraph` method

Both options clear it up quite a bit, but I like the method approach since it encapsulates the logic to the `IrGraph`. I'm happy with either way though.

Reply:
I kept it simple and went with your `computeServiceTypeReferenceInfo` suggestion.